src/coreclr/jit/fginline.cpp

```diff
@@ -47,7 +47,9 @@ unsigned Compiler::fgCheckInlineDepthAndRecursion(InlineInfo* inlineInfo)
             // This inline candidate has the same IL code buffer as an already
             // inlined method does.
             inlineResult->NoteFatal(InlineObservation::CALLSITE_IS_RECURSIVE);
-            break;
+
+            // No need to note CALLSITE_DEPTH we're already rejecting this candidate
+            return depth;
```

Comment:
otherwise we exit this loop and note one more "NoteFatal" while inliner in general don't expect from us multiple Fatal reasons, see https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/inlinepolicy.cpp#L187

Reply:
Did anything changed recently that caught this or is this a pre-existing bug?

Reply:
pre-existing one, it's super hard to reproduce in real world and it's absolutely harmless in Release.
In order to reproduce it without the test case you need Checked config and do 20 inlines (in deep) and 20th one should be recursive - almost impossible to meet in real world 🙂